This patch adds the requirements list to setup.py.
It might need to be updated for every different version of MDTF-diagnostics.
The version constraints are calculated from src/conda/env_base.yml.
Author: Stamen Miroslavov <stamen.miroslavov@bsc.es>
Barcelona Supercomputing Center 2026
diff -ruNBbw a/setup.py b/setup.py
--- a/setup.py	2026-10-21 15:27:58.795296265 +0200
+++ b/setup.py	2026-10-21 14:55:07.941153823 +0200
@@ -148,6 +148,22 @@
         'build_py': BuildPyWithAppResources,
         'develop': PostDevelopCommand,
         'install': PostInstallCommand
-    }
-    #   install_requires=[...] #external packages as dependencies
+    },
+    install_requires=[
+        'click>=8.1,<9',
+        'cftime>=1.6,<2',
+        'cf_xarray>=0.8.4',
+        'cfunits>=3.3.6,<4',
+        'dask>=2024.1.0',
+        'ecgtools>=2023.7.13',
+        'intake>=0.6.6,<2',
+        'intake-esm>=2024.2.6',
+        'matplotlib>=3.8',
+        'netCDF4>=1.6,<2',
+        'numpy>=1.26,<2',
+        'pandas>=2.1,<3',
+        'pint>=0.24',
+        'PyYAML>=6,<7',
+        'xarray>=2024.1',
+    ],
 )
